eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ImageConverter.php

```diff
@@ -163,6 +166,30 @@ public function toFieldValue(StorageFieldValue $value, FieldValue $fieldValue)
         $fieldValue->data = $this->parseLegacyXml($value->dataText);
     }
 
+    public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef): void
+    {
+        $validators = $fieldDef->fieldTypeConstraints->validators;
+
+        $storageDef->dataInt1 = $validators['FileSizeValidator']['maxFileSize'] ?? 0;
```

Comment:
don't you need to cast that as well?

Reply:
I believe it's a backport, so if there is a need for a type case, it should be added to original source as well.

Reply:
> don't you need to cast that as well? 

According to validation schema `$validators['FileSizeValidator']['maxFileSize']` is an integer value (or null) https://github.com/ezsystems/ezpublish-kernel/blob/15ca50fc5d96743080d9d96764531f8a7d7b5ea5/eZ/Publish/Core/FieldType/Image/Type.php#L25-L38